### PR TITLE
Add cosign for handling image functionality.

### PR DESCRIPTION
**Pull request by amartin120**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- Replaces default hauler image mechanics with cosign.
- Pulls in the once separated rancherfederal/OCIL library for the remaining oci functions not handled by cosign.

* **What is the current behavior?** (You can also link to an open issue here)
- When you add an image, that's it.  No other supported artifacts come with it.
- Does not work for multi-arch images.

* **What is the new behavior (if this is a feature change)?**
When you add an image to the store, it will also attempt to include any signatures/attestations.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: cmd/hauler/cli/store/add.go [hunk 1]
```diff
@@ -62,40 +62,48 @@ func storeFile(ctx context.Context, s *store.Layout, fi v1alpha1.File) error {
 type AddImageOpts struct {
 	*RootOpts
 	Name string
+	Key  string
 }
 
 func (o *AddImageOpts) AddFlags(cmd *cobra.Command) {
 	f := cmd.Flags()
-	_ = f
+	f.StringVarP(&o.Key, "key", "k", "", "(Optional) Path to the key for digital signature verification")
```
Comment (zackbradys) [inline cmd/hauler/cli/store/add.go:70]: match other description references
```suggestion
	f.StringVarP(&o.Key, "key", "k", "", "(Optional) Path to the key for image signature verification")
```

File: pkg/artifacts/file/getter/getter_test.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		},
+		{
+			name: "should identify a directory",
+			args: args{
+				source: rootDir,
+			},
+			want: "directory",
+		},
+		{
+			name: "should identify a http",
```
Comment (zackbradys) [inline pkg/artifacts/file/getter/getter_test.go:41]:
```suggestion
			name: "should identify an http fqdn",
```

File: pkg/artifacts/file/getter/getter_test.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+			args: args{
+				source: "http://my.cool.website",
+			},
+			want: "http",
+		},
+		{
+			name: "should identify a http",
```
Comment (zackbradys) [inline pkg/artifacts/file/getter/getter_test.go:48]:
```suggestion
			name: "should identify an http fqdn",
```

File: pkg/consts/consts.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+	// FileLayerMediaType is the reserved media type for File content layers
+	FileLayerMediaType = "application/vnd.content.hauler.file.layer.v1"
+
+	// FileLocalConfigMediaType is the reserved media type for File config
+	FileLocalConfigMediaType     = "application/vnd.content.hauler.file.local.config.v1+json"
+	FileDirectoryConfigMediaType = "application/vnd.content.hauler.file.directory.config.v1+json"
+	FileHttpConfigMediaType      = "application/vnd.content.hauler.file.http.config.v1+json"
+
+	// MemoryConfigMediaType
```
Comment (zackbradys) [inline pkg/consts/consts.go:31]: add description to "MemoryConfigMediaType is ..." to match others?

**Review by dweomer — APPROVED** [review]

**Pull request merged**